lib/elixir/test/elixir/code_test.exs

```diff
@@ -338,6 +342,21 @@ defmodule CodeTest do
                {[{{:x, :foo}, 2}], [x: :foo]}
     end
 
+    if :erlang.system_info(:otp_release) >= ~c"25" do
+      test "undefined function with custom env" do
+        env = Code.env_for_eval(CustomEnv.make_custom_env())
```

Comment:
```suggestion
        env = Code.env_for_eval(__ENV__)
```

This would be fine as well, no?

Reply:
In this case, we wouldn't be evaluating this since expansion [works differently](https://github.com/elixir-lang/elixir/blob/main/lib/elixir/src/elixir_expand.erl#L851-L852) when `env.function == nil`

Reply:
But inside the test we should have function set as well, no?

Reply:
Oh you're right, sorry 🤦 Will fix

Reply:
Fixed in [ed3b281](https://github.com/elixir-lang/elixir/pull/12998/commits/ed3b2812e3922e9fdf8ed3c0a27243b5b46bca7e)